llzk/tests/function_dialect.rs

```diff
@@ -85,3 +86,122 @@ fn function_call() {
 }";
     assert_eq!(ir, expected);
 }
+
+fn make_empty_struct<'c>(context: &'c LlzkContext, name: &str) -> StructDefOp<'c> {
+    let loc = Location::unknown(&context);
+    let typ = StructType::from_str(&context, name);
+    r#struct::def(loc, name, &[], {
+        [
+            r#struct::helpers::compute_fn(loc, typ, &[], None).map(Into::into),
+            r#struct::helpers::constrain_fn(loc, typ, &[], None).map(Into::into),
+        ]
+    })
+    .unwrap()
+}
+
+#[test]
+fn func_def_op_self_value_of_compute() {
+    common::setup();
+    let context = LlzkContext::new();
+    let module = llzk_module(Location::unknown(&context));
+    let module_body = module.body();
+
+    let s = make_empty_struct(&context, "StructA");
+    let s = StructDefOpRef::try_from(module_body.append_operation(s.into())).unwrap();
+    assert!(s.verify());
```

Comment:
You can use `verify_operation_with_diags` to get the diagnostics if anything goes wrong.

For example: 

```suggestion
    llzk::operation::verify_operation_with_diags(&s).expect("verification failed");
```